### PR TITLE
Updated api guide

### PR DESCRIPTION
**Pull request by nehaalhat**
@maelk @digambar15
kindly review.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -1,8 +1,88 @@
...
+The following is a sample CRD of a HardwareClassificationController resource (in YAML), it includes its metadata and specification section.
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: HardwareClassification
+metadata:
+  name: hardwareclassification-sample
+  namespace: metal3
+  labels:
+    hardwareclassification-sample: matches
```
Comment (maelk) [inline docs/api.md:91]: Is there any place where this, and the label set on the BMH, are explained ?

Comment (digambar15) [reply docs/api.md:69]: Good catch.  @nehaalhat we should add in above above explaination. How these labels added in BMH hosts and how we can see it.

Comment (digambar15) [reply docs/api.md:91]: @maelk  We can mention key and value to be added on BMH hosts on one note. That's ok I guess.

Comment (nehaalhat) [reply docs/api.md:91]: Added explanation for label parameter.
Commit: 12f5abaf4a802305ae4c068b6e2a0e871fd0b664
@digambar15 @maelk kindly review.

**Review by digambar15 — APPROVED** [review]
Looks good

File: docs/api.md (hunk 1 above)
Comment (digambar15) [reply docs/api.md:91]: Thanks!

**Pull request merged**